core/import.ts

```diff
@@ -0,0 +1,287 @@
+function importSubscribe(): void {
+  $feeds.set(
+    Array.from(
+      new Set(
+        $importedFeedsByCategory
+          .get()
+          .flatMap(([, feeds]) => feeds.map(feed => feed))
+      )
+    )
+  )
+
+  $categories.set(
+    Array.from(
+      new Set($importedFeedsByCategory.get().flatMap(([category]) => category))
+    )
+  )
+}
+
+export function selectAllImportedFeeds(): void {
+  $importedCategories.set($categories.get().map(category => category.id))
+  $importedFeeds.set($feeds.get().map(feed => feed.id))
+}
+
+export function clearImportSelections(): void {
+  $importedCategories.set([])
+  $importedFeeds.set([])
+}
+
+export function toggleImportedCategory(categoryId: string): void {
+  let selectedCategories = new Set($importedCategories.get())
+  let selectedFeeds = new Set($importedFeeds.get())
+
+  let feeds = $feeds.get().filter(feed => feed.categoryId === categoryId)
+
+  if (selectedCategories.has(categoryId)) {
+    selectedCategories.delete(categoryId)
+    feeds.forEach(feed => selectedFeeds.delete(feed.id))
+  } else {
+    selectedCategories.add(categoryId)
+    feeds.forEach(feed => selectedFeeds.add(feed.id))
+  }
+
+  $importedCategories.set(Array.from(selectedCategories))
+  $importedFeeds.set(Array.from(selectedFeeds))
+}
+
+export function toggleImportedFeed(feedId: string, categoryId: string): void {
+  let selectedCategories = new Set($importedCategories.get())
+  let selectedFeeds = new Set($importedFeeds.get())
+
+  if (selectedFeeds.has(feedId)) {
+    selectedFeeds.delete(feedId)
+    let remainingFeedsInCategory = $feeds
+      .get()
+      .filter(
+        feed => feed.categoryId === categoryId && selectedFeeds.has(feed.id)
+      )
+    if (remainingFeedsInCategory.length === 0) {
+      selectedCategories.delete(categoryId)
+    }
+  } else {
+    selectedFeeds.add(feedId)
+    selectedCategories.add(categoryId)
+  }
+
+  $importedCategories.set(Array.from(selectedCategories))
+  $importedFeeds.set(Array.from(selectedFeeds))
+}
+
+export function handleImportFile(file: File): Promise<void> {
+  return new Promise(resolve => {
+    $reading.set(true)
+    $importErrors.set([])
+    $importedFeedsByCategory.set([])
+    importSubscribe()
+    let reader = new FileReader()
+    reader.onload = async function (e) {
+      let content = e.target?.result as string
+      let fileExtension = file.name.split('.').pop()?.toLowerCase()
+
+      if (fileExtension === 'json') {
+        try {
+          let jsonData = JSON.parse(content)
+          if (jsonData.type === 'feedsByCategory') {
+            $importedFeedsByCategory.set(jsonData.data)
+            importSubscribe()
+            selectAllImportedFeeds()
+          } else {
+            $importErrors.set([
+              ...$importErrors.get(),
+              'Invalid JSON structure'
+            ])
+          }
+        } catch (jsonError) {
+          $importErrors.set(['Failed to parse JSON or invalid structure'])
```

Comment:
It will not support translation. We need to move to code and use I18n object in UI.